File: system-test/java/src/test/java/com/mycompany/myshop/systemtest/legacy/mod07/e2e/PlaceOrderPositiveTest.java
package com.mycompany.myshop.systemtest.legacy.mod07.e2e;

import com.mycompany.myshop.testkit.channel.ChannelType;
import com.mycompany.myshop.testkit.driver.port.dtos.OrderStatus;
import com.mycompany.myshop.systemtest.legacy.mod07.e2e.base.BaseE2eTest;
import com.optivem.testing.Channel;
import org.junit.jupiter.api.TestTemplate;

import static com.mycompany.myshop.systemtest.commons.constants.Defaults.*;

class PlaceOrderPositiveTest extends BaseE2eTest {
    @TestTemplate
    @Channel({ChannelType.UI, ChannelType.API})
    void shouldPlaceOrderForValidInput() {
        app.erp().returnsProduct().sku(SKU).unitPrice(20.00).execute()
                .shouldSucceed();

        app.myShop().placeOrder().orderNumber(ORDER_NUMBER).sku(SKU).quantity(5).country(COUNTRY).execute()
                .shouldSucceed()
                .orderNumber(ORDER_NUMBER)
                .orderNumberStartsWith("ORD-");

        app.myShop().viewOrder().orderNumber(ORDER_NUMBER).execute()
                .shouldSucceed()
                .orderNumber(ORDER_NUMBER)
                .sku(SKU)
                .quantity(5)
                .unitPrice(20.00)
                .status(OrderStatus.PLACED)
                .totalPriceGreaterThanZero();
    }
}

File: system-test/dotnet/SystemTests/Legacy/Mod07/E2eTests/PlaceOrderPositiveTest.cs
using Dsl.Core.UseCase;
using Driver.Port.Dtos;
using SystemTests.Commons.Constants;
using SystemTests.Legacy.Mod07.E2eTests.Base;
using Optivem.Testing;
using Xunit;

namespace SystemTests.Legacy.Mod07.E2eTests;

public class PlaceOrderPositiveTest : BaseE2eTest
{
    [Theory]
    [ChannelData(ChannelType.UI, ChannelType.API)]
    public async Task ShouldPlaceOrderForValidInput(Channel channel)
    {
        (await _app.Erp()
            .ReturnsProduct()
            .Sku(Defaults.SKU)
            .UnitPrice(20.00m)
            .Execute())
            .ShouldSucceed();

        var shop = await _app.MyShop(channel);
        (await shop.PlaceOrder()
            .OrderNumber(Defaults.ORDER_NUMBER)
            .Sku(Defaults.SKU)
            .Quantity(5)
            .Country(Defaults.COUNTRY)
            .Execute())
            .ShouldSucceed()
            .OrderNumber(Defaults.ORDER_NUMBER)
            .OrderNumberStartsWith("ORD-");

        (await shop.ViewOrder()
            .OrderNumber(Defaults.ORDER_NUMBER)
            .Execute())
            .ShouldSucceed()
            .OrderNumber(Defaults.ORDER_NUMBER)
            .Sku(Defaults.SKU)
            .Quantity(5)
            .UnitPrice(20.00m)
            .Status(OrderStatus.Placed)
            .TotalPriceGreaterThanZero();
    }
}
